frontend/packages/ceph-storage-plugin/src/components/create-storage-system/odf-external-providers/README.md

```diff
@@ -0,0 +1,182 @@
+
+```JS
+{
+    /* The product name of the external system to be displayed in the dropdown selection */
+    label: string;
+
+    /* The model which refers to the sub-system resource. */
+    model: ExtensionK8sModel;
+
+    /* 
+     * id is a unique identifier that will be used in naming StorageSystems
+     * and sub-StorageSystems e.g `odf-<name>-storage-system`.
+     */
+    id: string;
+
+    /* A React Functional Component to render the connection details of the external provider.*/
+    Component: React.FunctionComponent<ComponentProps>;
+
+    /* Handler function to create request payload for the external sub-StorageSystem */
+    createPayload: CreatePayload;
+};
+
+```
+
+### `label`
+The product name of the external system to be displayed in the dropdown selection of wizard.
+
+### `model`
+This is a reference to external vendor CRD. The model should be defined as:
+
+```js
+{
+    /* apiGroup of the external vendor CRD */
+    group: string;
+
+    /* apiVersion of the external vendor CRD */
+    version: string;
+
+    /* kind of the external vendor CRD */
+    kind: string;
+}
+```
+### `id`
+id is a unique identifier that will be used in naming StorageSystems and sub-StorageSystems e.g `odf-<name>-storage-system`.
+The name of id always be in small caps and should use hyphens if there are more than one word. We should avoid adding any additional prefix or suffix like `system` unless it is present in the name itself.
+e.g `ibm-flash`
+     
+### `Component`
+
+The Component to be rendered should only include the connection details e.g IP Address, username, password, etc. The Storage Class field is generic for all external Providers and is not required to be duplicated in this Component.
+
+* A new folder should be created for a provider as `./odf-external-provider/{id-of-provider}`. The name of the folder should be the same as `id`. [See`id`](#id).
+* Define your Component inside the newly created folder by adding a file with the same name as of the folder.
+* The Component must use the [Patternfly's form components](https://www.patternfly.org/v4/components/form) e.g `FormGroup`, `TextInput`, `Radio` to make the design consistent with the rest of the wizard and OpenShift console.
+* All form components should be controlled forms i.e using `onChange` event handler to control the value of the input elements.
+* The component should externalize the strings for i18n by namespace: `ceph-storage-plugin~`
+
+The Component has following three props:
+
+##### 1. `form`
+This is the state of the form that will be rendered by the Component. The state is a plain javascript object where each `key` represents a unique name of the form field and `value` represents the value of the field.
+
+For instance,
+
+```js
+   //  Intial render state
+   state = {};
+
+   // Post render state
+   state = {
+       username: 'kubeadmin',
+       password: 'gdjshdjshdX==',
+   }
```

Comment:
```suggestion
   state: FormState = {};

   // Post render state
   state = {
       username: 'kubeadmin',
       password: 'gdjshdjshdX==',
   }
```
Vendors should be able to leverage typescript. Make use of generics to provide vendors with support for typescript.